cloudinit/config/cc_resizefs.py

```diff
@@ -60,15 +60,28 @@ def _resize_btrfs(mount_point, devpth):
     if not util.mount_is_read_write(mount_point) and os.path.isdir(
         "%s/.snapshots" % mount_point
     ):
-        return (
+        cmd = [
             "btrfs",
             "filesystem",
             "resize",
             "max",
             "%s/.snapshots" % mount_point,
-        )
+        ]
     else:
-        return ("btrfs", "filesystem", "resize", "max", mount_point)
+        cmd = ["btrfs", "filesystem", "resize", "max", mount_point]
+
+    # btrfs has exclusive operations and resize may fail if btrfs is busy
+    # doing one of the operations that prevents resize. As of btrfs 5.10
+    # the resize operation can be queued
+    btrfs_with_queue = util.Version().from_str("5.10")
+    system_btrfs_ver = util.Version().from_str(
+        subp.subp(["btrfs", "--version"])[0].split("v")[-1].strip()
```

Comment:
fyi: `subp` has a named tuple return value for readability:

```python
subp.subp(["btrfs", "--version"]).stdout.split("v")[-1].strip()
```

The mocked return values would have to use `subp.SubpResult()` to pass, if you make this change.